### DISCUSSION TITLE
[Feature Request] Creality K1 Max Lidar support

### DISCUSSION THREAD

**Discussion by mwLabs-eu** [OrcaSlicer/OrcaSlicer#3125] [Ideas]
Hi,

just got my new printer, using Orca Slicer on it is working fine. But is it also possible to use the Lidar from it for flow calibration? When using Creality Slicer it prints 4 zig-zag lines and scans. Can't find anything similar in Orca Slicer.

If it's not available, is there any chance, this can be implemented in the future?

Thanks for helping :)

**Comment by Myklt8** [comment]
AI camera support as well

**Comment by Ltek** [comment]
put [Feature Request] on the thread title

**Reply by mwLabs-eu** [reply]
Done 👍🏻

**Comment by squimito** [comment]
I also would like these features if possible

**Comment by LightningShark25** [comment]
Would also like to know if this is doable.
I'm running root access on k1 max.  I'm not understanding where the limitation imposed by creality is - that feature only works with Creality Print - but why...?   If anyone understands the mechanism on the interaction I'd be curious to understand it.  It might help those looking at the feature

**Comment by Ltek** [comment]
according to this Creality post is maybe be accessible to Orca in the next FW update... https://www.reddit.com/r/crealityk1/comments/18wlsr7/upcoming_firmware_for_k1k1maxk1c/

**Reply by LightningShark25** [reply]
Thanks @Ltek that looks promising!
Do we think that feature will need to be added to the implementation of Orca to take advantage? 
I'm not clear where the functionality is coded - is it a gcode macro that requires calling from the slicer perhaps?

**Comment by Ltek** [comment]
@SoftFever you aware of this announcement by Creality?

**Comment by Myklt8** [comment]
I can confirm that you can send the file from Orca to the K1 and then print with calibration from the printer screen... although it overrides KAMP. Now we just need to be able to activate it directly from the slicer

**Reply by Ltek** [reply]
I think we want KAMP to run before Lidar runs its calibration.

**Comment by virology2407** [comment]
The new [firmware 1.3.3.5](https://www.crealitycloud.com/software-firmware/firmware/flagship-series) was released today, and it specifically indicates lidar support for Orca. Would be nice to see the feature implemented.

**Reply by SaturnsVoid** [reply]
The LiDAR functions can only be used from the printers screen, You have to enable them in the Camera Settings, then when you go to print you have to print from the printers screen and make sure to enable the "Calibration" option. The features seem to be coded into the OS of the printer, can't seem to make it call it from gcode, it seems to overide the "START_PRINT" macro too.

**Reply by Ltek** [reply]
Creality made a point to tell everyone that FW 1.3.3.5 exposes LiDAR to Orca, Cura, etc... it likely does.

**Reply by Myklt8** [reply]
It does work, but you have to download and initiate the print from the screen. What people want is an option like creality slicer has for 1 click printing with the calibration (if you want it)

**Reply by virology2407** [reply]
@SaturnsVoid The panel does indeed support files sliced with Orca. In previous versions of the firmware, it would display an error code saying the LiDAR function (or motion advance, whatever it is Creality brands it as) wasn't supported for the file. That's one piece of the puzzle.

The missing piece, and the request here, is to be able to trigger the process from Orca. Creality Print has an option to calibrate the printer that triggers this process remotely. What I've been doing as a workaround is slicing in Orca, uploading the GCODE, and triggering the print from Creality Print with the calibrate option enabled. I'd like to be able to drop Creality Print completely.

**Reply by dwaite** [reply]
Creality expanded compatibility with third party sliced G-Code (and kinda comically fixed some support in Creality Print). They didn't add anything to trigger LIDAR or AI Camera features via Klipper/G-Code. When you submit a print through one of the official mechanisms, it coordinates running those features outside Klipper, I believe modifying the G-Code just enough to make it work (e.g. adding a stop after the first layer completes)

**Comment by kanishk619** [comment]
Since I wasn't able to find anything regarding this, I did my own digging and found that you can connect to port 9999 with websocket and trigger print with AI support. Upload the file to printer and then trigger using below command.

```shell
echo '{"method":"set","params":{"opGcodeFile":"printprt:/usr/data/printer_data/gcodes/filename_to_be_printed.gcode","enableSelfTest":1,"aiPausePrint":1,"aiDetection":1,"aiFirstFloor":1,"aiSw":1}}' | ./websocat.x86_64-unknown-linux-musl ws://printer.ip.address:9999

```

Please note that this is on my rooted K1 Max using guilouz script. 
This works well with KAMP as well but need slicer start gcode changes like below 

> M140 S0
M104 S0 
RESPOND TYPE=command MSG="Starting Adaptive Bed Mesh..."
BED_MESH_CLEAR
BED_MESH_CALIBRATE
BED_MESH_PROFILE SAVE=adaptive
BED_MESH_PROFILE LOAD=adaptive
START_PRINT EXTRUDER_TEMP=[nozzle_temperature_initial_layer] BED_TEMP=[bed_temperature_initial_layer_single]

Hope to see this added to Orca :smiley:

**Reply by Thrashin** [reply]
> 
@kanishk619 
I don't really understand the steps on how to enable this feature without touching the screen do you think you could explain a little more im a noobie when it comes to web sockets, and Ip and other coding things. im good with technology just not that good. if possible could you maybe make a video on the steps you took, because orca slicer team hasn't taken any steps into integrating this into orca. secondly does this solution make it automatically use the lidar when i click print and upload or do i have to enter the name of the file in web sockets every time.

**Comment by Ltek** [comment]
If Orca added this K series owners would be in love with Orca

**Comment by virology2407** [comment]
This is a huge find. Hopefully the Orca team can implement this. 

On June 28, 2024, GitHub ***@***.***> wrote:
> Since I wasn't able to find anything regarding this, I did my own
> digging and found that you can connect to port 9999 with websocket and
> trigger print with AI support. Upload the file to printer and then
> trigger using below command.
>
> echo
> '{"method":"set","params":{"opGcodeFile":"printprt:/usr/data/printer_data/gcodes/filename_to_be_printed.gcode","enableSelfTest":1,"aiPausePrint":1,"aiDetection":1,"aiFirstFloor":1,"aiSw":1}}'
> | ./websocat.x86_64-unknown-linux-musl ws://printer.ip.address:9999
>
> Hope to see this added to Orca 😃
>

—
Reply to this email directly, view it on GitHub
<https://github.com/SoftFever/OrcaSlicer/discussions/3125#discussioncomment-
9904487>, or unsubscribe <https://github.com/notifications/unsubscribe-
auth/ARDKQF3JE55QDWIPPBSNRTLZJVH2NAVCNFSM6AAAAABAVSXSRCVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TSMBUGQ4DO>.
You are receiving this because you commented.Message ID:
***@***.***>

**Comment by dwaite** [comment]
Since this was originally opened, the Creality Print 5.x code base was released which is partially based on Orca. Have any interested parties evaluated the feasibility of a backport?

**Comment by fortellerq** [comment]
@dwaite 
Creality Print support was added recently in https://github.com/SoftFever/OrcaSlicer/pull/6928 so if I am connecting the dots correctly, right now adding support for LiDAR is just a matter of including proper parameters in startPrint function and handling that properly.

https://github.com/SoftFever/OrcaSlicer/blob/main/src/slic3r/Utils/CrealityPrint.cpp#187

**Comment by intothevoid1986** [comment]
Hi all!
I'm working to integrate the Creality AI feature in Orca.
Using wireshark I can validate the hypotesys of kanishk619 and - tweeking the code just a bit - I'm able to trigger a print with the "AI calibration" procedure using Orcaslicer.

From my analisys seem that the only required additional params is "enableSelfTest: 1", the others Ai params will be consequently set to 1.

My setup is rooted Creality K1 Max with stock firmware. Fluidd and GuppyScreen installed with Guilouz Helper Script, without removing Creality Web Service.

I will be happy to submit a Pull Request when finish the testing!

**Reply by shirogeek** [reply]
Let me know if I can help with the testing or whatsoever...

**Reply by fortellerq** [reply]
I could also help with testing, although I have exactly the same setup - K1 Max with Guilouz Helper Script installed. I could imagine that it would be worth testing on other Creality Print compatible printers if there aren't any unwanted behavior from having this option included in print request. May be worth to include as an option in some settings.

**Reply by deployor** [reply]
Updates?

**Reply by wallynm** [reply]
If you need any help with testing i can help aswell, sadly i dont know anything about printing hardware to help on the code.
Happy new year btw :P

**Reply by teamfighter** [reply]
+1 for that feature, it is highly needed.

**Reply by kkobold** [reply]
+1

**Comment by ArtaFakhari** [comment]
Any updates?😊

**Comment by snzsnz** [comment]
Bump

**Comment by dwaite** [comment]
I suspect this will not happen without someone to contribute code.
